### PR TITLE
Add connector for VocaDB / UtaiteDB / TouhouDB

### PR DESCRIPTION
**Pull request by sorusari343**
**Describe the changes you made**
Connector for https://vocadb.net/

**Additional context**

### CODE REVIEW

File: src/connectors/vocadb.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Connector.playButtonSelector = '.css-1lc7lii button[title="Play"]';
+
+Connector.getDuration = () => {
+	let store = getPlayQueueStore();
+	let current = store.items[store.currentIndex];
+	return current.entry.pvs[0].length;
+}
+
+function getPlayQueueStore() {
+	return JSON.parse(localStorage.PlayQueueStore);
```
Comment (jaccarmac) [inline src/connectors/vocadb.js:52]: Haven't tested this myself yet; Is there any performance hit from the constant `localStorage` access?

File: src/connectors/vocadb.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		default:
+			return artist;
+	}
+}
+
+Connector.playerSelector = '#app .css-1pm1wrk';
+Connector.trackArtSelector = '.css-1no5jxy';
+Connector.trackSelector = '.css-n3lbvk';
+Connector.artistSelector = '.css-molfmb';
+Connector.playButtonSelector = '.css-1lc7lii button[title="Play"]';
```
Comment (hornbuckle) [inline src/connectors/vocadb.js:43]: These randomly-generated React classes will likely change anytime an update is pushed to the site, which will break the connector. Will likely need to get more creative with how to target specific locations, or, if you have any input on this site, ask them to add static classes to those locations that can be used instead.

Comment (hornbuckle) [reply src/connectors/vocadb.js:43]: Assuming the player markup won't change when any other changes are pushed to the site, these selectors should work without using the randomly-generated classes.
```
Connector.playerSelector = '.container-fluid:not([class*=css-])';

Connector.playButtonSelector = `${Connector.playerSelector} .btn-group button .icon-play`;

Connector.artistSelector = `${Connector.playerSelector} > div > div:first-of-type a + div > a`;

Connector.trackSelector = `${Connector.playerSelector} > div > div:first-of-type div > span`;

Connector.trackArtSelector = `${Connector.playerSelector} > div > div:first-of-type a > div`;
```

File: src/connectors/vocadb.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const filter = MetadataFilter.createFilter({
+	artist: cleanupArtist,
+});
+
+Connector.applyFilter(filter);
+
+function cleanupArtist(artist) {
```
[the remark this answers is not in the document]
Comment (sorusari343) [reply src/connectors/vocadb.js:9]: As far as I know, this level of detail is unique to VocaDB because it is a specialist website.

**Pull request closed without merging**